pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+        "STATUS,RAMP", "RAMP,%dV/s",
+        """ A integer property that represents the ramp speed (V/s) of the output voltage of the
+        HV Source. This property can be set.""",
+        # When this property acts as get, the instrument will return a string like this:
+        # "RAMP, RANGE=3000V/s, VALUE=1000V/s", then voltage_ramp will return 1000.0,
+        # hence the convenience of the get_process.
+        validator=strict_range,
+        values=VOLTAGE_RAMP_RANGE,
+        get_process=lambda v:
+        float(re.findall(EurotestHPP120256.f_numbers_regex_pattern, v[2])[0])
+    )
+
+    measure_voltage = Instrument.measurement(
+        "STATUS,MU",
+        """ Measures the actual output voltage of the HV Source (kV).""",
+        # This property is a get so, the instrument will return a string like this:
+        # "U, RANGE=3.000kV, VALUE=2.458kV", then measure_voltage will return 2458.0,
+        # hence the convenience of the get_process.
+        get_process=lambda v:
+        float(re.findall(EurotestHPP120256.f_numbers_regex_pattern, v[2])[0])
+    )
+
+    voltage_range = Instrument.measurement(
+        "STATUS,MU",
+        """ Measures the actual output voltage range of the HV Source (kV).""",
+        # This property is a get so, the instrument will return a string like this:
+        # "U, RANGE=3.000kV, VALUE=2.458kV", then voltage_range will return 3000.0,
+        # hence the convenience of the get_process.
+        get_process=lambda v:
+        float(re.findall(EurotestHPP120256.f_numbers_regex_pattern, v[1])[0])
+    )
+
+    measure_current = Instrument.measurement(
+        "STATUS,MI",
+        """ Measures the actual output current of the power supply (mAmps).""",
+        # This property is a get so, the instrument will return a string like this:
+        # "I, RANGE=5000mA, VALUE=1739mA", then measure_current_range will return a 1739.0,
+        # hence the convenience of the get_process."""
+        get_process=lambda v:
+        float(re.findall(EurotestHPP120256.f_numbers_regex_pattern, v[2])[0])
+    )
+
+    current_range = Instrument.measurement(
+        "STATUS,MI",
+        """ Measures the actual output current range of the power supply (mAmps).""",
+        # This property is a get so, the instrument will return a string like this:
+        # "I, RANGE=5000mA, VALUE=1739mA, then current_range will return a 5000.0,
+        # hence the convenience of the get_process.
+        get_process=lambda v:
+        float(re.findall(EurotestHPP120256.f_numbers_regex_pattern, v[1])[0])
+    )
+
+    enable_kill = Instrument.setting(
+        "KILL,%s",
+        """ Enables or disables the kill function of the HV source.
+         When Kill is enabled yellow led is flashing and the output
+         will be shut OFF permanently without ramp if Iout > IOUTmax.""",
+        validator=strict_discrete_set,
+        values={'ON': 'ENable', 'OFF': 'DISable', 'ENable': 'ENable', 'DISable': 'DISable',
```

Comment:
I think it would be easier to leave only "ON"/"OFF" or better True/False choice to users.
All the other combinations are redundant.
Similar comments to other cases below.

Reply:
Agreed. A common pattern is to name such properties `kill_enabled` and use (only) True/False as the interface towards the users.

Reply:
Got it. Thanks